tests/test_main.py

```diff
@@ -27,3 +33,62 @@ def test_expand_abbreviations():
 
     expanded_template = expand_abbreviations(template, {})
     assert is_repo_url(expanded_template) is True
+
+
+@pytest.fixture(scope='session')
+def user_dir(tmpdir_factory):
+    return tmpdir_factory.mktemp('user_dir')
+
+
+@pytest.fixture(scope='session')
+def user_config_data(user_dir):
+    cookiecutters_dir = user_dir.mkdir('cookiecutters')
+    replay_dir = user_dir.mkdir('cookiecutter_replay')
+
+    return {
+        'cookiecutters_dir': str(cookiecutters_dir),
+        'replay_dir': str(replay_dir),
+        'default_context': {}
+    }
+
+
+@pytest.fixture(scope='session')
+def user_config_file(user_dir, user_config_data):
+    config_file = user_dir.join('config')
+    config_file.write(json.dumps(user_config_data))
+    return str(config_file)
+
+
+@pytest.fixture
+def template_url():
+    return 'https://github.com/pytest-dev/cookiecutter-pytest-plugin.git'
+
+
+@pytest.fixture
+def output_dir(tmpdir):
+    return tmpdir.mkdir('output')
+
+
+def test_cookiecutter_repository_url_should_clone(
+        mocker, template_url, output_dir, user_config_file, user_config_data):
+    mock_clone = mocker.patch(
+        'cookiecutter.main.clone',
```

Comment:
It seems like it would be clearer to patch `cookiecutter.vcs.clone` rather than the imported copy of it here.